third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -1105,46 +1107,47 @@ impl InterpreterImpl<'_> {
 
     /// Loads a resource from the data store and return the number of bytes read from the storage.
     fn load_resource<'c>(
-        data_store: &'c mut TransactionDataCache,
+        data_cache: &'c mut TransactionDataCache,
         resource_resolver: &impl ResourceResolver,
         module_storage: &impl ModuleStorage,
         gas_meter: &mut impl GasMeter,
         addr: AccountAddress,
         ty: &Type,
     ) -> PartialVMResult<&'c mut GlobalValue> {
-        match data_store.load_resource(module_storage, resource_resolver, addr, ty) {
-            Ok((gv, load_res)) => {
-                if let Some(bytes_loaded) = load_res {
-                    gas_meter.charge_load_resource(
-                        addr,
-                        TypeWithRuntimeEnvironment {
-                            ty,
-                            runtime_environment: module_storage.runtime_environment(),
-                        },
-                        gv.view(),
-                        bytes_loaded,
-                    )?;
-                }
-                Ok(gv)
-            },
-            Err(e) => Err(e),
+        if !data_cache.contains_resource(&addr, ty) {
+            let (entry, bytes_loaded) =
```

Comment:
can we have a small method somewhere that takes care of charing the bytes_loaded and just returning the entry that we insert in data_cache? (not sure if TransactionDataCache::load_resource can do that already)